applications/RomApplication/tests/test_thermal_lspg_rom.py

```diff
@@ -0,0 +1,100 @@
+import os
+import types
+import numpy as np
+
+import KratosMultiphysics
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+import KratosMultiphysics.kratos_utilities as kratos_utilities
+import KratosMultiphysics.RomApplication.rom_testing_utilities as rom_testing_utilities
+if kratos_utilities.CheckIfApplicationsAvailable("ConvectionDiffusionApplication"):
+    import KratosMultiphysics.ConvectionDiffusionApplication
+
+@KratosUnittest.skipIfApplicationsNotAvailable("ConvectionDiffusionApplication")
+class TestThermalLSPGRom(KratosUnittest.TestCase):
+
+    def setUp(self):
+        self.relative_tolerance = 1.0e-12
+
+    def testConvDiffStationaryLSPGRom2D(self):
+        self.work_folder = "thermal_static_test_files/LSPGROM/"
+        parameters_filename = "ProjectParametersLSPGROM.json"
+        expected_output_filename = "ExpectedOutputLSPGROM.npy"
+
+        with KratosUnittest.WorkFolderScope(self.work_folder, __file__):
+            # Set up simulation
+            with open(parameters_filename,'r') as parameter_file:
+                parameters = KratosMultiphysics.Parameters(parameter_file.read())
+            model = KratosMultiphysics.Model()
+            self.simulation = rom_testing_utilities.SetUpSimulationInstance(model, parameters)
+
+            # Run test case
+            self.simulation.Run()
+
+            # Check results
+            expected_output = np.load(expected_output_filename)
+            obtained_output = rom_testing_utilities.GetScalarNodalResults(self.simulation._GetSolver().GetComputingModelPart(), KratosMultiphysics.TEMPERATURE)
+            nodal_area = rom_testing_utilities.GetNodalAreaVector(self.simulation._GetSolver().GetComputingModelPart())
+
+            l2 = np.sqrt((sum(nodal_area*((1 - obtained_output/expected_output )**2)))/(sum(nodal_area)))*100
+            self.assertLess(l2, self.relative_tolerance)
+
+    def testConvDiffDynamicRom2D(self):
+        self.work_folder = "thermal_dynamic_test_files/LSPGROM/"
+        parameters_filename = "ProjectParametersLSPGROM.json"
+        expected_output_filename = "ExpectedOutputLSPGROM.npy"
+
+        time_snapshots = [500,1200,2500,3000,3600]
+
+        with KratosUnittest.WorkFolderScope(self.work_folder, __file__):
+            # Set up and run simulation
+            with open(parameters_filename,'r') as parameter_file:
+                parameters = KratosMultiphysics.Parameters(parameter_file.read())
+            model = KratosMultiphysics.Model()
+            self.simulation = rom_testing_utilities.SetUpSimulationInstance(model, parameters)
+
+            # Patch the RomAnalysis class to save the selected time steps results
+            def Initialize(cls):
+                super(type(self.simulation), cls).Initialize()
```

Comment:
```suggestion
                super().Initialize()
```

Reply:
Similar to previous discussion, it is necessary.